ultimate_cron.module

```diff
@@ -805,75 +418,34 @@ function ultimate_cron_get_hooks($reset = FALSE) {
 
   $hooks = array();
   // Generate list of jobs provided by modules.
-  $modules = module_list();
+  $modules = array_keys(Drupal::moduleHandler()->getModuleList());
   foreach ($modules as $module) {
     $hooks += ultimate_cron_get_module_hooks($module);
   }
 
   // Generate list of jobs provided by plugins.
-  $hooks += ultimate_cron_get_plugin_hooks();
-
+  // @todo: Port to Drupal 8
+  // $hooks += ultimate_cron_get_plugin_hooks();
   // Add default values to hooks.
-  ultimate_cron_prepare_hooks($hooks);
+   ultimate_cron_prepare_hooks($hooks);
 
   // Allow other to manipulate the hook list.
-  drupal_alter('cron', $hooks);
+  Drupal::moduleHandler()->alter('cron', $hooks);
 
   // Keep track on when we first registered new cron jobs.
   // This is used for as a base for time comparison in behind schedule
   // calculation for jobs that haven't run.
-  $registered = variable_get('ultimate_cron_hooks_registered', array());
   $new = array();
   foreach ($hooks as $name => $hook) {
     $new[$name] = empty($registered[$name]) ? REQUEST_TIME : $registered[$name];
   }
-  if ($registered != $new) {
-    variable_set('ultimate_cron_hooks_registered', $new);
-  }
 
   $cache = $hooks;
-  cache_set('ultimate_cron_hooks', $cache);
+  Drupal::cache()->set('ultimate_cron_hooks', $cache);
   return $cache;
 }
 
 // ---------- CRON RULE FUNCTIONS ----------
-
-/**
- * Form element validator for cron rules.
- */
-function ultimate_cron_plugin_crontab_element_validate_rule($element, \Drupal\Core\Form\FormStateInterface $form_state) {
```

Comment:
we need to make sure that we still have the validation for this in the cron job settings form.